apps/archives/views.py

```diff
@@ -259,7 +259,24 @@ def process_advanced_search(search_params):
     """
 
     docs_qs = Document.objects  # 'qs' for queryset
- 
+    keyword = search_params.get('keyword')
+    if keyword:
+        people_objs = Person.objects.filter(Q(last__contains=keyword) |
```

Comment:
This is a good start, but a few things:
- let's use `icontains` instead of `contains` for case insensitive search

- the search through Person.last and Person.first is currently searching for the full contents of the keyword field—which might be multiple words (maybe we should call it `keywords` instead of `keyword`?). This means if the keyword is "Foo Bar" and Person.last is "Bar", it won't match. See how that's implemented for searches on the `author` field.

Reply:
got it! I've renamed keyword and done as you suggested for author, and turned icontains to iexact (you don't want "hi" to return "Philip Morse")